frontend/src/main/java/olumofe/raphael/frontend/CarController.java

```diff
@@ -0,0 +1,63 @@
+package olumofe.raphael.frontend;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import starter.project.domain.Car;
+
+import java.util.*;
+
+
+@RestController
+public class CarController {
+
+    private List<Car> carNewList = new ArrayList<>();
+    Map<String, Car> map = new HashMap<>();
+
+    @PostMapping(path = "/car", consumes = {"application/json"})
+    public List<Car> addCar(@RequestBody Car car) {
+        carNewList.add(car);
+        return carNewList;
+    }
+
+    @GetMapping(path = "/car/{carId}")
+    public Car getCar(@PathVariable("carId") int carId) {
+
+        for (Car car : carNewList) {
+            if (car.getId() == (carId)) {
+                return car;
+            } else {throw new ResponseStatusException(
+                    HttpStatus.NOT_FOUND, "entity not found");
+            }
+        }
+        return null;
+    }
+
+    @GetMapping(path = "/cars/{list}")
+    public List<Car> getBatchCars(@PathVariable List list) {
+
+        List<Car> newList = new ArrayList<>();
+        for (Map.Entry<String, Car> m : map.entrySet()) {
```

Comment:
I am unsure what this Map is doing, can you please explain?

Reply:
its going through the map and if list contains a key in the map the value of that key being added to the list and returned

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#getOrDefault-java.lang.Object-V-